### ISSUE TITLE
Feature Request: Add support for AudioMoth metadata extraction

### ISSUE THREAD

**Issue by mbsantiago** [QutEcoacoustics/emu#349]
Hi Emu team!

Hello Emu team,

I recently developed a python package called [Metamoth](https://github.com/mbsantiago/metamoth), which extracts and parses the AudioMoth comment from the WAV header. It's able to extract all the relevant info from the comment, such as device ID, datetime, gain config, etc., and works on different AudioMoth firmware versions.

I think this feature could be very helpful for users of Emu who work with AudioMoth devices. Therefore, I would like to request that you consider adding support for AudioMoth metadata extraction in Emu.

I'd be happy to help with a pull request if you believe it could be helpful. Please let me know if you have any questions or concerns.

Thank you for your work on Emu!

Best regards,
Santiago

**Comment by atruskie** [comment]
I saw your Metamoth project. It's great work, good job.

EMU has a wider scope than that, so yeah, I would absolutely love to add support. Basically no one has asked me yet.

I see you've got a wide range of firmware support; do you also have fixtures (example files) for all those versions?

That's been my biggest blocker so far - getting a decent collection of AudioMoth files.

C# is an unusual choice in the academic community. I choose it because it is what I know, and I believe it makes for a maintainable solution. If you want to give it a try with a PR then great! Otherwise gathering fixtures would also be a great help.

Randomly: why is firmware 1.5.0 not supported?

**Comment by atruskie** [comment]
@mbsantiago I was just sitting down to scope this:

a) wow you did a lot of work - I like how you infer firmware versions from comment versions
b) the audiomoth comment structure is a nightmare

**Comment by mbsantiago** [comment]
Hey @atruskie, apologies for the delay in my response. I completely agree with you that the comment structure in Audiomoth is a bit messy 😖. To parse the comments, I used regular expressions, and I think they can be easily ported to C#. Would it be helpful if I shared the set of regex I used for each firmware version with you?

Initially, I thought of recording test samples for each firmware version of AudioMoth, but I haven't had the time to do so, and it seems quite tedious. Instead, I created Python fixtures to simulate comments from different versions, but it might be better to extract the comment generator function from the source code, if possible, to make it more reliable. These fixtures can also generate a bunch of random comments per firmware version for you to use as test cases, if you are interested.

Regarding firmware version 1.5.0, I skipped it as it wasn't related to the project I was working on, which involved analyzing recordings from versions 1.2.0 and 1.6.0. My aim was to implement the first firmware versions, but I lost momentum and skipped version 1.5.0.

Cheers
